siocli/lib/SqlDump.cpp

```diff
@@ -0,0 +1,796 @@
+    std::uint64_t m_trid;
+    std::uint64_t m_columnDefinitionId;
+};
+
+struct ColumnDefInfo {
+    std::uint64_t m_trid;
+    std::uint64_t m_columnId;
+};
+
+struct TableInfo {
+    std::uint64_t m_trid;
+    std::uint64_t m_currentColumnSetId;
+
+    std::string m_name;
+    std::vector<ColumnInfo> m_columns;
+};
+
+std::string constraintToString(const ColumnConstaint& constraint)
+{
+    // Only NOT NULL constraint is supported for now
+    if (constraint.m_type != ConstraintType::kNotNull) return std::string();
+
+    return getConstaintTypeName(constraint.m_type);
+}
+
+std::string formSelectCoreBody(const char* databaseName, const char* tableName,
+        const std::vector<std::string>& columnNames)
+{
+    std::stringstream ss;
+    ss << "SELECT ";
+
+    for (auto i = 0u; i < columnNames.size() - 1; ++i)
+        ss << columnNames[i] << ',';
+
+    if (!columnNames.empty()) ss << columnNames.back();
+
+    ss << " FROM " << databaseName << '.' << tableName;
+
+    return ss.str();
+}
+
+std::string formSelectCoreBody(const std::string& databaseName, const std::string& tableName,
+        const std::vector<std::string>& ColumnNames)
+{
+    return formSelectCoreBody(databaseName.c_str(), tableName.c_str(), ColumnNames);
+}
+
+std::string formSelectAllQuery(const std::string& databaseName, const std::string& tableName)
+{
+    std::stringstream ss;
+    ss << "SELECT * FROM " << databaseName << '.' << tableName;
+    return ss.str();
+}
+
+std::string formCreateDatabaseQuery(const DatabaseInfo& dbInfo)
+{
+    std::stringstream ss;
+    ss << "CREATE DATABASE " << dbInfo.m_name;
+
+    if (!dbInfo.m_cipherId.empty())
+        ss << " WITH " << kSysDatabases_CipherId_ColumnName << " = '" << dbInfo.m_cipherId << '\'';
+
+    // We can't create a seed value from database key
+    // ss << ", " << kSysDatabases_CipherKey_ColumnName << " = " << dbInfo.m_cipherKey;
+
+    return ss.str();
+}
+
+std::string formCreateTableQuery(const std::string& databaseName, const TableInfo& tableInfo)
+{
+    std::stringstream ss;
+
+    ss << "CREATE TABLE " << databaseName << '.' << tableInfo.m_name;
+
+    if (!tableInfo.m_columns.empty()) {
+        ss << " (";
+
+        for (auto i = 0u; i < tableInfo.m_columns.size() - 1; ++i) {
+            const auto& column = tableInfo.m_columns[i];
+            ss << " " << column.m_name << " " << getColumnDataTypeName(column.m_dataType);
+            for (const auto& constraint : column.m_constraints)
+                ss << " " << constraintToString(constraint);
+
+            ss << ',';
+        }
+
+        const auto& column = tableInfo.m_columns.back();
+        ss << " " << column.m_name << " " << getColumnDataTypeName(column.m_dataType);
+        for (const auto& constraint : column.m_constraints)
+            ss << " " << constraintToString(constraint);
+
+        ss << " )";
+    }
+
+    return ss.str();
+}
+
+BinaryValue readBinaryValue(google::protobuf::io::CodedInputStream& input)
```

Comment:
Please create class named as `CustomCodedInputStream` in the `common/lib/siodb/common/protobuf` and put there all these `readXXX()` methods.